File: src/main/java/me/diax/diax/commands/music/Queue.java
package me.diax.diax.commands.music;

import me.diax.comportment.jdacommand.Command;
import me.diax.comportment.jdacommand.CommandDescription;
import me.diax.diax.music.GuildMusicManager;
import me.diax.diax.music.MusicTrack;
import me.diax.diax.util.Emote;
import me.diax.diax.util.StringUtil;
import net.dv8tion.jda.core.entities.Message;

import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;

@CommandDescription(
        name = "queue",
        triggers = "queue",
        description = "Displays the songs in the queue."
)
public class Queue implements Command {

    @Override
    public void execute(Message message, String s) {
        GuildMusicManager manager = GuildMusicManager.getManagerFor(message.getGuild());
        List<MusicTrack> queue = new ArrayList<>(manager.getScheduler().getQueue());
        queue.add(0, manager.getScheduler().getCurrentTrack());
        String msg;
        if (queue.isEmpty() || manager.getScheduler().getCurrentTrack() == null) {
            msg = Emote.MUSICAL_NOTE + " - The queue is empty.";
        } else {
            List<MusicTrack> reducedQueue = queue.subList(0, queue.size() > 10 ? 10 : queue.size());
            msg = reducedQueue.stream().map(track -> "` (" + (reducedQueue.indexOf(track) + 1) + ") " + StringUtil.stripMarkdown(track.getTrack().getInfo().title) + " | " + StringUtil.stripMarkdown(track.getRequester().getNickname()) + "#" + track.getRequester().getUser().getDiscriminator() + "`").collect(Collectors.joining("\n", "Queue:\n\n", String.format("\n\nShowing (%s/%s) items.", reducedQueue.size(), queue.size())));
        }
        message.getChannel().sendMessage(msg).queue();
    }
}

File: src/main/kotlin/me/diax/diax/commands/music/Queue.kt
package me.diax.diax.commands.music

import me.diax.comportment.jdacommand.Command
import me.diax.comportment.jdacommand.CommandDescription
import me.diax.diax.music.GuildMusicManager
import me.diax.diax.util.Emote.MUSICAL_NOTE
import me.diax.diax.util.StringUtil
import net.dv8tion.jda.core.entities.Message
import java.util.*
import java.util.stream.Collectors

@CommandDescription(name = "queue", triggers = arrayOf("queue"))
class Queue : Command {

    override fun execute(message: Message, s: String) {
        val manager = GuildMusicManager.getManagerFor(message.guild)
        val queue = ArrayList(manager.scheduler.queue)
        queue.add(0, manager.scheduler.currentTrack)
        val msg: String
        when (queue.isEmpty() || manager.scheduler.currentTrack == null) {
            true -> msg = "The queue is empty."
            false -> {
                val reducedQueue = queue.subList(0, if (queue.size > 10) 10 else queue.size)
                msg = reducedQueue.stream().map { track -> "` (${if (reducedQueue.indexOf(track) == 9) "" else "0"}${reducedQueue.indexOf(track) + 1}) ${StringUtil.stripMarkdown(track.track.info.title)} | ${StringUtil.stripMarkdown(track.track.info.author)} | ${StringUtil.stripMarkdown(track.requester.effectiveName)}#${track.requester.user.discriminator} `" }.collect(Collectors.joining("\n", "Queue:\n\n", String.format("\n\nShowing (%s/%s) items.", reducedQueue.size, queue.size)))
            }
        }
        message.channel.sendMessage("$MUSICAL_NOTE - $msg").queue()
    }
}